### DISCUSSION TITLE
On-Demand Revalidation for new content

### DISCUSSION THREAD

**Discussion by dev-fredericfox** [vercel/next.js#35849] [Help] [answered]
I have a blog and use on-demand revalidation. It works like a charm when modifying an article.
However I can't get it to work when I post a new article. The index page show the new article, but the link to the article dosen't work (404) and I always get a 500 error. Does anyone have any hints how to get it to work?

Here is my code snippet, where Path refers to the edited blog post or the path of the new blog post.

When I post a new article I run validate for index ('/') and once more for the new post slug ('/post/19' for example)

```ts
import type { NextApiRequest, NextApiResponse } from "next";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  const path = req.body.pathToRevalidate
  
  // Check for secret to confirm this is a valid request
  if (req.query.secret !== process.env.REVALIDATE_SECRET) {
    console.log({ message: "Invalid token" })
    return res.status(401).json({ message: "Invalid token" });
  }

  try {
    await res.unstable_revalidate(path);
    return res.json({ revalidated: true });
  } catch (err) {
    // If there was an error, Next.js will continue
    // to show the last successfully generated page
    return res.status(500).send(err);
  }
}


```

**Comment by dev-fredericfox** [comment, marked as answer]
Ok so I figured out my error, I was trying to revalidate a "new" page, but that's actually not necessary. Instead I needed to set my `getStaticPaths()` fallback to 'blocking' instead of `false`. Hope this will help someone else learning about next.js!

**Reply by yamanidev** [reply]
how did you deal with slugs that are invalid? Do you programmatically return 404 pages?

**Comment by stormsson** [comment]
@dev-fredericfox this works , but has a (wanted ?) side effect: each request for a new page (or, broadly speaking a non-previously built page) will interact with your page source, for example a headless cms.

There are situations where you do not want this behavior because  having a regeneration+API call occur from input of the end user  could lead to exhaust  the quota of the data source (if any)
